Sources/IndexStore/Index Store.md

```diff
@@ -0,0 +1,192 @@
+
+### Record files
+
+We already saw an example of a record file’s contents in the *Overview* section. We will thus not dive any deeper into its contents.
+
+While unit files represent an entire compilation unit, record files represent the contents of individual source files. For the most part these are `.swift`, `.c`. `.h` and similar files but inside the SDK also `.swiftinterface` files may be indexed and thus generate a record file.
+
+Each record file contains a hash of its contents at the end of the file name. Consider a particular record file as the interpretation of its contents during a compilation. As long as all declarations and references remain the same and at the same location, a source file’s record doesn’t change. This means that eg. adding content to an end of line comment doesn’t change the record. If, however, a declaration is modified, an overload resolves differently, or any other significant modification is made, then the source file will have a different record.
+
+## Miscellaneous
+
+### Incremental builds
+
+At the moment, the Index Store is only ever appended to and no unit or record files are every removed. Apart from using disk space, this does not pose an issue for record files if only those record files referenced by unit files are iterated. The Index Store may, however, contain stale unit files:
+1. A source file was deleted from the project but the unit file of its last build is still present in the Index Store. This can be easily checked for by ensuring a unit’s main file exists on disk.
+2. A source file used to be part of an iOS and watchOS target but was removed from the watchOS target and then modified. The outdated unit file of the last build in the watchOS target still exist in the Index Store and the check in (1) will not catch this because the source file also still exists. SourceKit-LSP solves this issue by querying the build system for the output paths it produces and filtering unit files based on their output path. Depending on the complexity of your project, it may be acceptable to not consider this niece use case. Otherwise a similar integration with the build system is necessary.
+
+None of this is a consideration if the Index Store was created by a clean build because no old unit or record files exist in this case.
+
+### Multiple definitions of the same symbol
+
+Note that a symbol may be defined more than once, eg. in the following examples.
+
+- `foo()` is defined in line 2 when compiling for Windows but in line 4 for all other platforms
+```swift
+#if os(Windows)
+func foo() { ... }
+#else
+func foo() { ... }
+#endif
+```
+- Different targets that have a `main` function will all have separate definitions of those `main` functions. This would result in a linker issue if these functions were linked into the same binary but since each `main` function is in its own binary, there is no conflict.
```

Comment:
I'm confused with what section is for. Do you want to describe what's the implication of this for _Index Store_ ?